File: .tmp_py4013/batch_12_dump/3350_adjacent_increasing_subarrays_detection_ii.js
// LeetCode 3350 - Adjacent Increasing Subarrays Detection II
// https://leetcode.com/problems/adjacent-increasing-subarrays-detection-ii/

function ok(up, n, k) {
    for (let i = 0; i + 2 * k <= n; i++) {
        if (up[i] >= k && up[i + k] >= k) return true;
    }
    return false;
}
var maxIncreasingSubarrays = function(nums) {
    const n = nums.length;
    const up = new Array(n);
    up[n - 1] = 1;
    for (let i = n - 2; i >= 0; i--) {
        up[i] = (nums[i] < nums[i + 1]) ? up[i + 1] + 1 : 1;
    }
    let lo = 1, hi = Math.floor(n / 2);
    while (lo < hi) {
        const mid = Math.floor((lo + hi + 1) / 2);
        if (ok(up, n, mid)) lo = mid;
        else hi = mid - 1;
    }
    return lo;
};

File: .tmp_rs4013/rust_check16/3350_adjacent_increasing_subarrays_detection_ii.rs
struct Solution;
// LeetCode 3350 - Adjacent Increasing Subarrays Detection II
// https://leetcode.com/problems/adjacent-increasing-subarrays-detection-ii/

impl Solution {
    pub fn max_increasing_subarrays(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut up = vec![0; n];
        up[n - 1] = 1;
        for i in (0..n - 1).rev() {
            up[i] = if nums[i] < nums[i + 1] { up[i + 1] + 1 } else { 1 };
        }
        let mut lo = 1;
        let mut hi = n / 2;
        let ok = |k: usize| -> bool {
            let mut i = 0;
            while i + 2 * k <= n {
                if up[i] >= k as i32 && up[i + k] >= k as i32 {
                    return true;
                }
                i += 1;
            }
            false
        };
        while lo < hi {
            let mid = (lo + hi + 1) / 2;
            if ok(mid) {
                lo = mid;
            } else {
                hi = mid - 1;
            }
        }
        lo as i32
    }
}

fn main() {}
